container_images/tox/README.md

```diff
@@ -0,0 +1,63 @@
+A prow worker that executes Python tests using [tox](https://tox.readthedocs.io/).
+Check the Dockerfile for which Python versions are installed.
+
+### Interface
+
+Your job:
+1. Create a [tox configuration file](https://tox.readthedocs.io/en/latest/config.html) in your
+repository's test directory.
+   * Optionally, write test artifacts to the `ARTIFACTS` directory.
+1. Configure prow with the test directory's path as the first positional argument.
+
+Prior to invoking, the runner will:
+1. Checkout your repository, mount it as a volume, and set its root as Docker's 
+[workdir](https://docs.docker.com/engine/reference/commandline/run/#set-working-directory--w).
+2. Mount a volume for artifacts, and inject its path with the `ARTIFACTS` env variable.
+
+### Example
+
+Given a repository called `project` in the user's home directory:
+
+```shell script
+~/project
+├── bin
+├── docs
+└── src
+    └── py
+        └── tests
+            └── tox.ini
+```
+
+This tox configuration will execute [pytest](https://docs.pytest.org/) for both Python 3.6 and Python 3.8,
+and ensure that results are written to the `ARTIFACTS` directory as junit XML files.
+
+```ini
+[tox]
+envlist = py36, py38
+
+[testenv]
+deps =
+    pytest
+
+commands = pytest --junit-prefix={envname} \
```

Comment:
usually we would just make a container image which runs this command, rather than one that runs tox and expects a tox config which may have this command. this puts us back in the position of potentially dangerous PRs - we have to know a tox config exists, and that it was not amended during the PR. 

is the main advantage that it will run for multiple python runtimes?

Reply:
I'm not familiar with the potentially dangerous PRs situation -- what did you have in mind?

Reply:
ah, the way we've built the others is that they don't take any command configuration from the repo to be tested, and they're strictly separated. so i was wondering what the advantage we get from this, so we could understand pros/cons

Reply:
since you've outlined that in the other comment, maybe we ship the tox config into the container and don't read it from the repo?

Reply:
Updated  to generate the tox.ini file at test-execution time.